lib/analist/checker.rb

```diff
@@ -64,22 +62,29 @@ def check_send(node) # rubocop:disable Metrics/PerceivedComplexity
     end
 
     def significant_difference?(annotation, other_annotation)
-      attrs = %i[receiver_type args_types return_type]
-      attrs.delete(:args_types) if [annotation, other_annotation].any? do |a|
-        a.args_types.any? { |t| t == Analist::Annotation::TypeUnknown } ||
-        a.args_types == [Analist::Annotation::AnyArgs]
+      return if [annotation.args_types, other_annotation.args_types].any? do |a|
```

Comment:
It may be better to split this method into multiple different methods for each of the cases. That could make it more clear how each of the `if` cases is different from the others.